File: solid-kata/java/lsp-solution/src/main/java/com/codurance/lsp/ElectricCar.java
package com.codurance.lsp;

public class ElectricCar extends Vehicle implements ElectricityPowered {
    private static final int BATTERY_FULL = 100;
    private int batteryLevel;

    @Override
    public void chargeBattery() {
        batteryLevel = BATTERY_FULL;
    }

    public int batteryLevel() {
        return batteryLevel;
    }
}

File: solid-kata/c#/LSP/ElectricCar.cs
namespace LSP
{
    public class ElectricCar : Vehicle
    {
        private const int BatteryFull = 100;
        public int BatteryLevel { get; private set; }

        public override void FillUpWithFuel()
            => throw new InvalidOperationException("It's an electric car");

        public override void ChargeBattery() => BatteryLevel = BatteryFull;
    }
}